### PR TITLE
update merge_request comments API to support line and path

### PR DESCRIPTION
**Pull request by zlx**
#7813

### CODE REVIEW

**Comment by giedriusr** [comment]
Any news on when this is going to be merged?

**Comment by zlx** [comment]
I have rebase this MR. extremely hope it can be merged.

**Comment by Razer6** [comment]
@randx @dosire Can you take a look?. It's fresh, it's mergable :)

**Comment by dosire** [comment]
@Razer6 I can't merge any code, my skills are weak. I'm pretty good at answering email though :)

**Comment by Razer6** [comment]
@dosire Grab new opportunities ;-) Now, this MR is one step further to implement services like Hound-CI for gitlab :)

**Comment by dosire** [comment]
@Razer6 :) It would be great to have this, but it should really be reviewed by @maxlazio or @randx

**Comment by zlx** [comment]
@Razer6 @dosire  I'm working on a project call [Gitlab-Hound](https://github.com/zlx/Gitlab-Hound)

File: lib/api/merge_requests.rb [hunk 1]
```diff
@@ -234,7 +237,27 @@ def handle_merge_request_errors!(errors)
...
+            if diff_line
+              opts[:line_code] = Gitlab::Diff::LineCode.generate(
+                diff_file.new_path,
+                diff_line.new_pos,
+                diff_line.old_pos
+              )
+              break
+            end
+          end
+        end
```
Comment (randx) [inline lib/api/merge_requests.rb]: This code is really complicated. Can you refactor it for easier review?

File: spec/requests/api/merge_requests_spec.rb [hunk 2]
```diff
@@ -358,14 +361,27 @@
 
   describe "POST /projects/:id/merge_request/:merge_request_id/comments" do
     it "should return comment" do
-      post api("/projects/#{project.id}/merge_request/#{merge_request.id}/comments", user), note: "My comment"
-      expect(response.status).to eq(201)
-      expect(json_response['note']).to eq('My comment')
+      post api(merge_request_comments_url, user), note: "My comment"
```
Comment (zlx) [inline spec/requests/api/merge_requests_spec.rb]: :terrified Why it make me to use single-quoted after I change to use double-quoted follow the comments?

**Comment by jvanbaarsen** [comment]
This merge request has been closed because a request for more information has not been reacted to for more than 2 weeks. If you respond and conform to the merge request guidelines in our [contributing guidelines](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CONTRIBUTING.md#pull-requests) we will reopen this merge request.

**Pull request closed without merging**